src/Persistence_matrix/include/gudhi/matrix.h

```diff
@@ -0,0 +1,2106 @@
+{
+  if constexpr (!PersistenceMatrixOptions::is_z2) {
+    if (colSettings_->operators.get_characteristic() != static_cast<characteristic_type>(-1)) {
+      std::cerr << "Warning: Characteristic already initialised. Changing it could lead to incoherences in the matrice "
+                   "as the modulo was already applied to values in existing columns.";
+    }
+
+    colSettings_->operators.set_characteristic(characteristic);
+  }
+}
+
+template <class PersistenceMatrixOptions>
+template <class Container_type>
+inline void Matrix<PersistenceMatrixOptions>::insert_column(const Container_type& column) 
+{
+  if constexpr (!PersistenceMatrixOptions::is_z2){
+    GUDHI_CHECK(colSettings_->operators.get_characteristic() != static_cast<characteristic_type>(-1),
+                std::logic_error("Matrix::insert_column - Columns cannot be initialized if the coefficient field "
+                                 "characteristic is not specified."));
+  }
+
+  static_assert(
+      !isNonBasic,
+      "'insert_column' not available for the chosen options. The input has to be in the form of a face boundary.");
+  matrix_.insert_column(column);
+}
+
+template <class PersistenceMatrixOptions>
+template <class Container_type>
+inline void Matrix<PersistenceMatrixOptions>::insert_column(const Container_type& column, index columnIndex) 
+{
+  if constexpr (!PersistenceMatrixOptions::is_z2){
+    GUDHI_CHECK(colSettings_->operators.get_characteristic() != static_cast<characteristic_type>(-1),
+                std::logic_error("Matrix::insert_column - Columns cannot be initialized if the coefficient field "
+                                 "characteristic is not specified."));
+  }
+
+  static_assert(!isNonBasic && !PersistenceMatrixOptions::has_column_compression,
+                "'insert_column' with those parameters is not available for the chosen options.");
+  static_assert(!PersistenceMatrixOptions::has_row_access,
+                "Columns have to be inserted at the end of the matrix when row access is enabled.");
+  matrix_.insert_column(column, columnIndex);
+}
+
+template <class PersistenceMatrixOptions>
+template <class Boundary_type>
+inline typename Matrix<PersistenceMatrixOptions>::insertion_return_type
+Matrix<PersistenceMatrixOptions>::insert_boundary(const Boundary_type& boundary, dimension_type dim)
+{
+  if constexpr (!PersistenceMatrixOptions::is_z2){
+    GUDHI_CHECK(colSettings_->operators.get_characteristic() != static_cast<characteristic_type>(-1),
+                std::logic_error("Matrix::insert_boundary - Columns cannot be initialized if the coefficient field "
+                                 "characteristic is not specified."));
+  }
+
+  if constexpr (isNonBasic && !PersistenceMatrixOptions::is_of_boundary_type &&
+                PersistenceMatrixOptions::column_indexation_type == Column_indexation_types::CONTAINER)
+    return matrix_.insert_boundary(boundary, dim);
+  else
+    matrix_.insert_boundary(boundary, dim);
```

Comment:
If `matrix_.insert_boundary(boundary, dim);` returns void exactly in the right cases, we could just write `return matrix_.insert_boundary(boundary, dim);` without need for 2 cases.
(if sometimes `matrix_.insert_boundary(boundary, dim);` returns a vector but we don't want to return that vector, then forget this comment)